src/core/weightedGraph2.js

```diff
@@ -37,89 +65,252 @@ export type WeightedEdge = {|
+   *
+   * Clients must not modify the graph during iteration. If they do so, an
+   * error may be thrown at the iteration call site.
+   *
+   * The weightedEdges are yielded in sorted address order.
+   */
+  edges(options: EdgesOptions): Iterator<WeightedEdge> {
+    return this._edgesIterator(options);
+  }
+
+  *_edgesIterator(options: EdgesOptions): Iterator<WeightedEdge> {
+    const edges = this._graph.edges(options);
+    for (const edge of edges) {
+      const weight = this.getEdgeAddressWeight(edge.address);
+      yield {edge, weight};
+    }
   }
 
   /**
    * Return the weight of a Node address.
    */
-  getNodeAddressWeight(_unused_a: NodeAddressT): NodeWeight {
-    throw new Error("method not implemented");
+  getNodeAddressWeight(n: NodeAddressT): NodeWeight {
+    return this._weights.getNodeWeight(n);
   }
 
   /**
    * Return the weight of an Edge address.
    */
-  getEdgeAddressWeight(_unused_a: EdgeAddressT): EdgeWeight {
-    throw new Error("method not implemented");
+  getEdgeAddressWeight(e: EdgeAddressT): EdgeWeight {
+    return this._weights.getEdgeWeight(e);
   }
 
   // Utilities
 
-  merge(_unused_wg: WeightedGraph2): WeightedGraph2 {
-    throw new Error("method not implemented");
+  /**
+   * Produce a copy of this WeightedGraph.
+   *
+   * The copy is equal to the original, but distinct, so that they may be
+   * modified independently.
+   */
+  copy(): WeightedGraph2 {
+    return WeightedGraph2.merge([this]);
   }
 
-  compareWeights(_unused_wg: WeightedGraph2): WeightsComparison {
-    throw new Error("method not implemented");
+  /**
+   * Compute the union of the given WeightedGraphs. The result is a new
+   * WeightedGraph that has all of the nodes and all of the edges from all the
+   * inputs.
+   *
+   * If two of the given graphs have edges with the same address, the edges
+   * must be equal (i.e. must have the same source and destination in each
+   * graph). If this is not the case, an error will be thrown.
+   *
+   * Example usage:
+   *
+   * let wg1 = new WeightedGraph().addNode(a).addNode(b).addEdge(e);
+   * let wg2 = new WeightedGraph().addNode(b).addNode(c).addEdge(f);
+   * let wg3 = WeightedGraph.merge([g1, g2]);
+   * Array.from(wg3.nodes()).length;  // 3
+   * Array.from(wg3.edges()).length;  // 2
+   * wg1 = new WeightedGraph().addNode(a).addNode(b).addEdge(x);
+   * wg2 = new WeightedGraph().addNode(c);
+   * wg3 = WeightedGraph.merge([g1, g2]);
+   *
+   * The newly created WeightedGraph is a separate instance from any of the
+   * input WeightedGraphs, and may be mutated independently.
+   */
+  static merge(wgs: Iterable<WeightedGraph2>): WeightedGraph2 {
+    const graphs: Array<Graph> = [];
+    const weights: Array<WeightsI> = [];
+    for (const wg of wgs) {
+      graphs.push(wg._graph);
+      weights.push(wg._weights);
+    }
+    return new WeightedGraph2(Graph.merge(graphs), Weights().merge(weights));
   }
```

Comment:
```js
// This seems cleaner to me
const graphs = wgs.map({_graph} => _graph)
const weights = wgs.map({_weights} => _weights)
```
```js
// If you're feeling frisky
const graphs = wgs.map(prop("_graph"))
const weights = wgs.map(prop("_weights"))
```

Reply:
@elihanover  I considered this but at the risk of being pedantic, this syntax is not valid because `Iterable`s do not have access to array methods. 

Moreover, this suggestion requires two full iterations over the `Iterable`, instead of one, and I wanted to optimize this a bit.

Reply:
Gotcha, makes sense to me.